doc/musig.md

```diff
@@ -0,0 +1,87 @@
+
+**Phase 1: pubnonce generation:** A PSBT is sent to the signing device, and it does not contain any pubnonce.
+- If a session already exists, it is deleted from the persistent memory.
+- A new session is created in volatile memory.
+- The device produces a fresh random number $rand\_root$, and saves it in the current session.
+- The device generates the randomness for the $i$-th input and for the $j$-th key as: $rand_{i,j} = SHA256(rand\_root || i || j)$.
+- Compute each *(secnonce, pubnonce)* as per the `NonceGen` algorithm.
+- At completion (after all the pubnonces are returned), the session secret $rand\_root$ is copied into the persistent memory.
+
+**Phase 2: partial signature generation:** A PSBT containing all the pubnonces is sent to the device.
+- *A copy of the session is stored in the volatile memory, and the session is deleted from the persistent memory*.
+- For each input/musig-key pair $(i, j)$:
+  - Recompute the pubnonce/secnonce pair using `NonceGen` with the synthetic randomness $rand_{i,j}$ as above.
+  - Verify that the pubnonce contained in the PSBT matches the one synthetically recomputed.
+  - Continue the signing flow as per BIP-0327, generating the partial signature.
+
+### Security considerations
+#### State reuse avoidance
+Storing the session in persistent memory only at the end of Phase 1, and deleting it before beginning Phase 2 simplifies auditing and making sure that there is no reuse of state across signing sessions.
+
+#### Security of synthetic randomness
+
+Generating $rand_{i, j}$ synthetically is not a problem, since the $rand\_root$ value is kept secret and never leaves the device. This ensures that all the values produced for different $i$ and $j$ not predictable for an attacker.
+
+#### Malleability of the PSBT
+If the optional parameters are passed to the _NonceGen_ function, they will depend on the transaction data present in the PSBT. Therefore, there is no guarantee that they will be unchanged the next time the PSBT is provided.
+
+However, that does not constitute a security risk, as those parameters are only used as additional sources of entropy in _NonceGen_. A malicious software wallet can't affect the _secnonce_/_pubnonce_ pairs in any predictable way. Changing any of the parameters used in _NonceGen_ would cause a failure during Phase 2, as the recomputed _pubnonce_ would not match the one in the psbt.
+
+### Generalization to multiple PSBT signing sessions
+
+The approach described above assumes that no attempt to sign a PSBT containing for a wallet policy containing `musig()` keys is initiated while a session is already in progress.
```

Comment:
This part was actually incomplete, as it doesn't say how the actual implementation works (just claiming it can be generalized). Added a short description in 64897008ab6643c1cb698fcc6b9abb771f9e4540.